packages/backend/src/accounting/psql/ledger-transfer/index.ts

```diff
@@ -38,3 +65,135 @@ export async function getAccountTransfers(
     { credits: [], debits: [] } as GetTransfersResult
   )
 }
+
+export async function createTransfers(
+  deps: ServiceDependencies,
+  transfers: CreateTransferArgs[],
+  trx?: TransactionOrKnex
+): Promise<CreateTransfersResult> {
+  const validationResults = await Promise.all(
+    transfers.map((transfer) =>
+      validateTransfer(deps, {
```

Comment:
I think we'd need some sort of locking to really enforce balance limits.

I'd previously had the thought that it might be worth considering storing the balance(s) on the accounts for easier lookup but it'd also make locking easier here

Reply:
Yeah I've been thinking about this, I think we will need to provide a `FOR KEY UPDATE` lock on the credit& debitAccountIds

Reply:
I'm cautious about de-normalizing and storing balances on the accounts since having a discrepancy between the sum(transfers) vs the calculated balances is scary

Reply:
We can always move to calculated balances with a lot of scale as well

Reply:
> I'm cautious about de-normalizing and storing balances on the accounts since having a discrepancy between the sum(transfers) vs the stored calculated balances is scary

:100:

Reply:
Going to leave locking for follow-up, going to focus on getting tests right for the balances first